src/semgrep_agent/semgrep.py

```diff
@@ -107,61 +110,112 @@ def rewrite_sarif_file(sarif_path: Path) -> None:
         json.dump(sarif_results, sarif_file, indent=2, sort_keys=True)
 
 
+@contextmanager
+def fix_head_for_github(meta: GitMeta) -> Iterator[Optional[str]]:
+    """
+    GHA can checkout the incorrect commit for a PR (it will create a fake merge commit),
+    so we need to reset the head to the actual PR branch head before continuing.
+
+    Note that this code is written in a generic manner, so that it becomes a no-op when
+    the CI system has not artifically altered the HEAD ref.
+
+    :return: The baseline ref as a commit hash
+    """
+
+    stashed_rev: Optional[str] = None
+    base_ref: Optional[str] = meta.base_commit_ref
+
+    if get_git_repo() is None:
+        yield base_ref
+        return
+
+    if base_ref:
+        # Preserve location of head^ after we possibly change location below
+        base_ref = git(["rev-parse", base_ref]).stdout.decode("utf-8").rstrip()
+
+    if meta.head_ref:
+        stashed_rev = git(["branch", "--show-current"]).stdout.decode("utf-8").rstrip()
```

Comment:
`git branch --show-current` doesn't work for me. Is this relying on a git feature people might not have?

Reply:
Since this only applies in GHA, I think it is not; since it works in our GHA environment it should be sufficient (also, wow, I think you need to update your git!)